bgpd/bgp_route.c

```diff
@@ -4310,6 +4310,9 @@ void bgp_clear_stale_route(struct peer *peer, afi_t afi, safi_t safi)
 
 int bgp_outbound_policy_exists(struct peer *peer, struct bgp_filter *filter)
 {
+	if (peer->sort == BGP_PEER_IBGP)
+		return 1;
+
 	if (peer->sort == BGP_PEER_EBGP
```

Comment:
Isn't this check already taking care of EBGP? All the OR condition are clubbed and explicit EBGP check is present to ensure this is not applied to IGBP. Why is extra check above this required?

Reply:
Never mind I think I missed the fact that you are returning true in case of IBGP.

Reply:
yes